tests/QA/test_radio.py

```diff
@@ -46,57 +42,52 @@ def test_bandwidth_big_packets(self, dev: conftest.BCDevice):
         requirement = conftest.get_requirement('radio.bwbig')
         assert(bandwidth(dev.link_uri, requirement['packet_size']) > requirement['limit_low'])
 
-    
     @pytest.mark.requirements("syslink_flowctrl")
     def test_reliability(self, dev: conftest.BCDevice):
         requirement = conftest.get_requirement('radio.reliability')
         # The bandwidth function will assert if there is any packet loss
         bandwidth(dev.link_uri, 4, requirement['limit_low'])
 
 
-def build_data(i, packet_size):
-    assert(packet_size % 4 == 0)
-    repeats = packet_size // 4
-    return struct.pack('<' + 'I' * repeats, *[i] * repeats)
+def latency(uri, timeout=10):
+    """
+    Retrieve the latency to a Crazyflie.
 
+    Args:
+        uri (str): The URI of the Crazyflie.
+        timeout (float): Maximum time to wait for latency updates.
 
-def latency(uri, packet_size=4, count=500):
-    link = cflib.crtp.get_link_driver(uri)
+    Returns:
+        float: The latency value received.
 
-    try:
-        pk = CRTPPacket()
-        pk.set_header(CRTPPort.LINKCTRL, 0)  # Echo channel
+    Raises:
+        TimeoutError: If the timeout is reached during latency retrieval.
+    """
+    with ValidatedSyncCrazyflie(uri) as scf:
```

Comment:
Just a question. We already have a Crazyflie object here in the BcDevice that we get from the test fixture. 
Should we not send that object to the ValidatedSyncCrazyflie so we do not create a new object on the same uri?